euouae_check.py

```diff
@@ -0,0 +1,88 @@
+# Accepts as arguments file or directory paths,
+# checks all gabc files containing antiphons
+# for standard EUOUAE lyrics and tones.
+
+import os
+import os.path
+import sys
+
+import arpeggio
+from music21 import converter, volpiano
+from music21.text import assembleLyrics
+import chant21
+from chant21.gabc.converter import AlterationWarning
+
+euouae_tones = {}
+
+
+class Skip(RuntimeError):
+    pass
+
+
+def _check(filename, verbose):
+    if not filename.endswith('.gabc'):
+        raise Skip('not a gabc')
+    if '_' in filename:
+        raise Skip('submitter (not selected) version')
+
+    if verbose:
+        print('checking {}'.format(filename))
+
+    try:
+        chant = converter.parse(filename)
+    except (arpeggio.NoMatch, AlterationWarning):
+        raise Skip('chant21 cannot parse')
+
+    metadata = chant.editorial.metadata
+    office_part = metadata.get('office-part')
+    if office_part != 'an':
+        raise Skip('not an antiphon ({})'.format(office_part))
+
+    errors = []
+
+    last_section = chant.sections[-1]
+    lyrics = assembleLyrics(last_section)
+    has_euouae = 'E U O U A E.' in lyrics
+    # TODO: is there a rule which ones have EUOUAE and which Et sic finiatur?
```

Comment:
EUOUAE is replaced with Et sic finiatur if and only if the GP is omitted at the end of the psalm, which occurs only in Tenebrae (and other hours in the Triduum) and in O.Def.
There are antiphons that I had to duplicate (they are soft-linked through the database website of nocturnale.marteo.fr) because they are used in a context with GP and in another without (e.g. Tenebrae antiphons also used in Our Lady of Sorrows).